clang/test/Driver/sycl-bitfield-accesses.cpp

```diff
@@ -0,0 +1,42 @@
+// RUN: %clangxx -O0 -fsycl-device-only %s -S -emit-llvm -o- | FileCheck %s
+
+// CHECK: %struct.with_bitfield = type { i32, i32, i32, i32 }
+//
+// Tests if fine grained access for SPIR targets is working
+
+struct with_bitfield {
+    unsigned int a : 32;
+    unsigned int b : 32;
+    unsigned int c : 32;
+    unsigned int d : 32;
+};
+
+#include "sycl.hpp"
+
+using namespace sycl;
+
+int main() {
+  sycl::queue queue;
+  std::vector<unsigned int> vec(1);
+  {
+    sycl::buffer<unsigned int> buf(vec.data(), vec.size());
+    queue.submit([&](sycl::handler &cgh) {
+      sycl::accessor acc(buf, cgh, sycl::write_only, sycl::no_init);
+      cgh.single_task<>([=]() {
+        with_bitfield A;
```

Comment:
Can we remove all this boilerplate by turning `A` to an SYCL_EXTERNAL function parameter?
This way optimization level configuration is optional.

```c++
SYCL_EXTERNAL unsigned int foo(with_bitfield A) {
  return A.a + A.b + A.c + A.d;
}
```

Reply:
That's a good idea. i made those changes. thanks.